File: services/catcher/pkg/document/domain/models/document.go
package models

import (
	"time"

	"github.com/google/uuid"
)

//Agragar transaccion anterior (Padre) // Id de documento

type Document struct {
	ID                string    `bson:"_id" json:"id"`    //Id del documento
	Type              string    `bson:"type" json:"type"` //D(document), T(exto)
	Name              string    `bson:"name" json:"name"` //name of the document
	ParentID          string    `bson:"parent_id" json:"parent_id"`
	UserID            string    `bson:"user_id" json:"user_id"`               //id del creador del documento
	TransactionId     string    `bson:"transaction_id" json:"transaction_id"` //id del documento una vez estando en el blockchain
	TransactionResult string    `bson:"transaction_result" json:"transaction_result"`
	Status            string    `bson:"status" json:"status"`
	Data              string    `bson:"data" json:"data"`         // data a registrar en el blockchain
	Platform          string    `bson:"platform" json:"platform"` // que plataforma se uso para registrar en el blockchain
	UplodedDate       time.Time `bson:"upload_date" json:"upload_date"`
	CreationDate      time.Time `bson:"creation_date" json:"creation_date"`
}

func (d *Document) New(client_id string, typeOfDocument string) Document {
	d.ID = uuid.New().String()
	d.CreationDate = time.Now()
	d.UplodedDate = time.Now()
	d.UserID = client_id
	d.Type = typeOfDocument
	return *d
}

type Documents []*Document

File: services/checker/src/domain/models/Document.ts

export class Document implements IDocument {
    public user_id: string;
    public data: string;
    public transaction_id: string;
    public status: string;
    public upload_date: Date;
    public creation_date: Date;
    public _id?: any


    constructor(user_id: string, data: string, transaction_id: string, status: string, upload_date: Date, creation_date: Date, id?: any) {
        this.user_id = user_id;
        this.data = data;
        this.transaction_id = transaction_id;
        this.status = status
        this.upload_date = upload_date
        this.creation_date = creation_date
        this._id = id
    }
}


export interface IDocument {
    user_id: string;
    data: string;
    transaction_id: string;
    status: string;
    upload_date: Date;
    creation_date: Date;
    _id?: any
}
